CMakeLists.txt

```diff
@@ -87,6 +87,71 @@ endif() # TRITON_ENABLE_GPU
 
+add_custom_command(
+  OUTPUT
+    pytorch_libs/libmkl_core.so
+    pytorch_libs/libmkl_gnu_thread.so
+    pytorch_libs/libmkl_intel_lp64.so
+    pytorch_libs/libmkl_intel_thread.so
+    pytorch_libs/libmkl_def.so
+    pytorch_libs/libmkl_avx2.so
+    pytorch_libs/libiomp5.so
+    pytorch_libs/libc10.so
+    pytorch_libs/libc10_cuda.so
+    pytorch_libs/libtorch.so
+    pytorch_libs/libtorch_cpu.so
+    pytorch_libs/libtorch_cuda.so
+    pytorch_libs/libtorchvision.so
+    LICENSE.pytorch
+    include/torch
+    include/torchvision
+  COMMAND ${CMAKE_COMMAND} -E make_directory "pytorch_libs"
+  COMMAND ${CMAKE_COMMAND} -E make_directory "include/torchvision"
+  COMMAND docker pull ${TRITON_PYTORCH_DOCKER_IMAGE}
+  COMMAND docker rm pytorch_backend_ptlib || echo "error ignored..." || true
+  COMMAND docker create --name pytorch_backend_ptlib ${TRITON_PYTORCH_DOCKER_IMAGE}
+  COMMAND docker cp pytorch_backend_ptlib:/opt/conda/lib/libmkl_core.so pytorch_libs/libmkl_core.so 
+  COMMAND docker cp pytorch_backend_ptlib:/opt/conda/lib/libmkl_gnu_thread.so pytorch_libs/libmkl_gnu_thread.so 
+  COMMAND docker cp pytorch_backend_ptlib:/opt/conda/lib/libmkl_intel_lp64.so pytorch_libs/libmkl_intel_lp64.so 
+  COMMAND docker cp pytorch_backend_ptlib:/opt/conda/lib/libmkl_intel_thread.so pytorch_libs/libmkl_intel_thread.so 
+  COMMAND docker cp pytorch_backend_ptlib:/opt/conda/lib/libmkl_def.so pytorch_libs/libmkl_def.so 
+  COMMAND docker cp pytorch_backend_ptlib:/opt/conda/lib/libmkl_avx2.so pytorch_libs/libmkl_avx2.so 
+  COMMAND docker cp pytorch_backend_ptlib:/opt/conda/lib/libiomp5.so pytorch_libs/libiomp5.so 
+  COMMAND docker cp pytorch_backend_ptlib:/opt/conda/lib/python3.8/site-packages/torch/lib/libc10.so pytorch_libs/libc10.so  
+  COMMAND docker cp pytorch_backend_ptlib:/opt/conda/lib/python3.8/site-packages/torch/lib/libc10_cuda.so pytorch_libs/libc10_cuda.so  
+  COMMAND docker cp pytorch_backend_ptlib:/opt/conda/lib/python3.8/site-packages/torch/lib/libtorch.so pytorch_libs/libtorch.so 
+  COMMAND docker cp pytorch_backend_ptlib:/opt/conda/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so pytorch_libs/libtorch_cpu.so 
+  COMMAND docker cp pytorch_backend_ptlib:/opt/conda/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so pytorch_libs/libtorch_cuda.so 
+  COMMAND docker cp pytorch_backend_ptlib:/opt/pytorch/vision/build/libtorchvision.so pytorch_libs/libtorchvision.so 
+  COMMAND docker cp pytorch_backend_ptlib:/opt/pytorch/pytorch/LICENSE LICENSE.pytorch
+  COMMAND docker cp pytorch_backend_ptlib:/opt/conda/lib/python3.8/site-packages/torch/include include/torch
+  COMMAND docker cp pytorch_backend_ptlib:/opt/pytorch/vision/torchvision/csrc include/torchvision/torchvision
+  COMMAND docker rm pytorch_backend_ptlib
+  COMMENT "Extracting pytorch and torchvision libraries and includes from ${TRITON_PYTORCH_DOCKER_IMAGE}"
+)
+add_custom_target(ptlib_target
```

Comment:
Can you combine these custom targets and following add_library, etc. together like this? The set_target_properties seems especially suspect.

Reply:
Followed similar to what is done in TF repo, added a comment to clarify it does what we want